From: Nick Rosbrook <enr0n@ubuntu.com>
Date: Tue, 16 Dec 2025 11:56:27 -0500
Subject: test: use journalctl -n option instead of piping to head

Bug-Ubuntu: https://launchpad.net/bugs/2136497

On Ubuntu's test infrastructure for the development series, this test
often fails when piping the journal output to head. The cause is
unclear, but possibly related to Ubuntu's use of uutils coreutils.

Workaround this by just using journalctl's -n flag, which removes the
need for piping output to head.
---
 test/units/TEST-04-JOURNAL.journal.sh | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/test/units/TEST-04-JOURNAL.journal.sh b/test/units/TEST-04-JOURNAL.journal.sh
index de3b45e..03613b5 100755
--- a/test/units/TEST-04-JOURNAL.journal.sh
+++ b/test/units/TEST-04-JOURNAL.journal.sh
@@ -69,12 +69,12 @@ journalctl -b -t "$ID" --truncate-newline | grep -v TAIL >/dev/null
 journalctl -b -1 -b all -m >/dev/null
 
 # -b always behaves like -b0
-journalctl -q -b-1 -b0 | head -1 >/tmp/expected
-journalctl -q -b-1 -b | head -1 >/tmp/output
+journalctl -q -b-1 -b0 -n+1 >/tmp/expected
+journalctl -q -b-1 -b -n+1 >/tmp/output
 diff /tmp/expected /tmp/output
 # ... even when another option follows (both of these should fail due to -m)
-{ journalctl -ball -b0 -m 2>&1 || :; } | head -1 >/tmp/expected
-{ journalctl -ball -b  -m 2>&1 || :; } | head -1 >/tmp/output
+{ journalctl -ball -b0 -n+1 -m 2>&1 || :; } >/tmp/expected
+{ journalctl -ball -b  -n+1 -m 2>&1 || :; } >/tmp/output
 diff /tmp/expected /tmp/output
 
 # https://github.com/systemd/systemd/issues/13708
